pulseapi/profiles/serializers.py

```diff
@@ -24,7 +24,27 @@ class Meta:
 class UserProfileSerializer(serializers.ModelSerializer):
     """
     Serializes a user profile.
+
+    Note that the following fields should only show up when
+    the 'enable_extended_information' flag is set to True:
+
+    - user_bio_long
+    - program_type
+    - program_year
+    - affiliation
     """
+
+    def __init__(self, instance, *args, **kwargs):
```

Comment:
`instance` shouldn't be a positional arg since it can be `None` (for e.g. during deserialization).
Here's what the `init` usually looks like for serializer classes: 
https://github.com/encode/django-rest-framework/blob/78367ba1024ec43349aba362758e826fdeafbb03/rest_framework/serializers.py#L111

you can probably just change it to:
```python
if instance is not None:
    if instance.enable_extended_information is False:
        self.fields.pop('user_bio_long')
        self.fields.pop('program_type')
        self.fields.pop('program_year')
        self.fields.pop('affiliation')
    # Whether this flag is set or not, it should not
    # end up in the actual serialized profile data.
    self.fields.pop('enable_extended_information')
```